### PR TITLE
integrate i18n-js with simualator

### PR DESCRIPTION
**Pull request by pavanjoshi914**
Fixes #

#### Describe the changes you have made in this PR -
Yarn dependencies
1> i18n-js
2> erb-loader

Setup is done with webpacker without using asset pipeline
### Screenshots of the changes (If any) -
Output - 
Screenshot from 2021-05-27 15-03-20

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by pavanjoshi914** [comment]
```en.js``` and``` ja.js``` are yml translations exported as js on client side

```i18n.js``` and ```translations.js``` are script files generated by i18n-js middleware configured in application.rb

File: app/views/layouts/simulator.html.erb [hunk 1]
```diff
@@ -4,6 +4,12 @@
     <title>CircuitVerse - Digital Circuit Simulator online</title>
     <%= csrf_meta_tags %>
     <script src="/js/pace.js"></script>
+    <%= javascript_include_tag "i18n" %>
+    <%= javascript_include_tag "translations", skip_pipeline: true %>
```
Comment (satu0king) [inline app/views/layouts/simulator.html.erb:8]: `javascript_include_tag` is a rails thing, we don't want to use this. But not sure.
CC @tachyons

Comment (pavanjoshi914) [reply app/views/layouts/simulator.html.erb:8]: I checked this what this tag is doing is creates html javascript tag ```<script src="/assets/i18n.23232323...."></script>```

same file is generated by middleware at ```/public/javascripts/i18n.js```  if i use ```<script src="/javascripts/i18n.js"></script>```
It's working perfectly fine

**Comment by pavanjoshi914** [comment]
@tachyons  
1. first dependency on rails side was ```<%= I18n::JS.filtered_translations.to_json %>;```
instead of this i am planning to use npm ```[yaml loader](https://www.npmjs.com/package/yaml-loader)``` and ```[json loader](https://www.npmjs.com/package/json-loader)``` for parsing .yaml files into json objects.

This will eliminate use of ```rails-erb-loader``` as well as use of index.js.erb file.

2. Passing locale value on frontend
   as we have to fetch locale from backend , so this can be done in the same way as shown in this pr.
   
   As a alternate we can put locale in url and parse it via creating a JavaScript method
   
Your opinion on this ?

**Comment by pavanjoshi914** [comment]
closing as we  shifted to other banana-i18n library for simulator

**Pull request closed without merging**